rflx/pyrflx/typevalue.py

```diff
@@ -260,3 +464,351 @@ def accepted_type(self) -> type:
+
+            self._last_field = nxt
+            nxt = self._next_field(nxt)
+
+    def get(self, fld: str) -> Any:
+        if fld not in self.valid_fields:
+            raise ValueError(f"field {fld} not valid")
+        return self._fields[fld].typeval.value
+
+    @property
+    def to_bitstring(self) -> Bitstring:
+        bits = ""
+        field = self._next_field(model.INITIAL.name)
+        while True:
+            if not field or field == model.FINAL.name:
+                break
+            field_val = self._fields[field]
+            if (
+                not field_val.set
+                or not isinstance(field_val.first, Number)
+                or not field_val.first.value <= len(bits)
+            ):
+                break
+            bits = bits[: field_val.first.value] + str(self._fields[field].typeval.to_bitstring)
+            field = self._next_field(field)
+        if len(bits) % 8:
+            raise ValueError(f"message length must be dividable by 8 ({len(bits)})")
+
+        return Bitstring(bits)
+
+    @property
+    def value(self) -> Any:
+        raise NotImplementedError
+
+    @property
+    def to_bytes(self) -> bytes:
+        bits = str(self.to_bitstring)
+        return b"".join(
+            [int(bits[i : i + 8], 2).to_bytes(1, "big") for i in range(0, len(bits), 8)]
+        )
+
+    @property
+    def fields(self) -> List[str]:
+        return [f.name for f in self._model.fields]
+
+    @property
+    def accessible_fields(self) -> List[str]:
+
+        nxt = self._next_field(model.INITIAL.name)
+        fields: List[str] = []
+        while nxt and nxt != model.FINAL.name:
+
+            if (
+                self.__simplified(self._model.field_condition(model.Field(nxt))) != TRUE
+                or not self._has_first(nxt)
+                or (
+                    not self._has_length(nxt)
+                    if not isinstance(self._fields[nxt].typeval, OpaqueValue)
+                    else self._is_valid_opaque_node(nxt)
+                )
+            ):
+                break
+
+            fields.append(nxt)
+            nxt = self._next_field(nxt)
+        return fields
+
+    def _is_valid_opaque_node(self, nxt: str) -> bool:
+
+        if self._get_length_unchecked(nxt) == UNDEFINED:
+            return True
+
+        for edge in self._model.incoming(model.Field(nxt)):
+            if self.__simplified(edge.condition) == TRUE:
+                valid_edge = edge
+                break
+        else:
+            return False
+
+        for ve in valid_edge.length.variables():
+            assert isinstance(ve.name, str)
+            if ve.name in self._fields and not self._fields[ve.name].set:
+                return True
+
+            if isinstance(ve, Last) and ve.name == "Message":
+                return False
+
+        return False
+
+    @property
+    def valid_fields(self) -> List[str]:
+        return [
+            f
+            for f in self.accessible_fields
```

Comment:
If a list comprehension does not fit into one line, black splits it in three parts (expression, `for`, `if`) and puts each part on one line.

Reply:
I was aware of the if condition but I wasn't aware that black splits it specifically into those three parts. I probably wouldn't have annotated it if the first part was not just `f`. Since this seems to be imposed by black I'd keep it as is.